keras/utils/data_utils.py

```diff
@@ -351,6 +351,12 @@ def __len__(self):
         """
         raise NotImplementedError
 
+    @abstractmethod
+    def on_epoch_end(self):
```

Comment:
It is and it isn't.
The `shuffle` arg relates to the shuffling of the batch indices in the sequences, however, you may also wish to implement file path shuffling in the Sequence method. This would allow that by placing that shuffle operator into the on_epoch_end method.
The path shuffling would ensure that the files inside the batches differed as well as the batch ordering.